inventories/test/routeros_hosts.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/python3
+import json
+import subprocess
+import locale
+import os
+import stat
+
+# Only usable if the identity file has the hostname in it.
+class VagrantIdentityFiles:
+    """Caches all Vagrant SSH IdentityFile locations for fast lookup"""
+
+    def __init__(self):
+        completed = subprocess.run(['/bin/sh', '-c', 'vagrant ssh-config | grep -F IdentityFile'],
+                                   capture_output=True, encoding=locale.getpreferredencoding())
+        #print(f'vagrant returned {completed}')
+        self.__identities = self.__extract_identityfiles_only(completed.stdout.splitlines())
+        #print(f'identities are {self.__identities}')
+
+    def __extract_identityfiles_only(self, identityfile_list):
+        out = []
+        for identity in identityfile_list:
+            out.append(identity.strip()[len('IdentityFile '):])
+        return out
+
+    def get_identityfile_or_empty(self, host):
+        #print(f'get_identityfile for {host}')
+        for identity in self.__identities:
+            if f'/{host}/' in identity:
+                print(f'get_identityfile found {identity}')
+                return identity
+        if len(self.__identities) == 1:
+            return self.__identities[0]
+        return ''
+
+
+# Alternative for use when using a single common identity file.
+class LocalIdentityFile:
+    """ Always returns the same local identity file """
+
+    def __init__(self, relative_filename):
+        self.__identityfile = os.path.join(os.path.abspath(os.path.dirname(__file__)),
+                                           relative_filename)
+        os.chmod(self.__identityfile, stat.S_IRUSR | stat.S_IWUSR)
+
+    def get_identityfile_or_empty(self, host):
+        return self.__identityfile
+
+
+def make_hostvars(hosts, identities, common_vars):
+    hostvars = {}
+    for host in hosts.keys():
+        identityfile = identities.get_identityfile_or_empty(host)
+        #print(f'make_hostvars - identityfile is {identityfile}')
+        if not identityfile:
+            continue
+        tmp = {}
+        tmp.update(common_vars)
+        tmp["instance"] = host
+        tmp["ansible_host"] = hosts[host]
+        tmp["ansible_ssh_private_key_file"] = identityfile
+        hostvars[host] = tmp;
+    return hostvars
+
+
+if __name__ == "__main__":
+    identities = LocalIdentityFile("/home/dpk/.vagrant.d/insecure_private_key")
```

Comment:
this looks specific to your setup, is this how you mean to do this? perhaps this is the line that belongs commented out with a note that this is an example for other people to adapt for their environment while line 67 gets uncommented?

Reply:
This is sort of a glue record for ansible and vagrant. Let me update this to use HOME environment variable instead to make it more portable.